libraries/Wire/src/Wire.cpp

```diff
@@ -205,13 +288,18 @@ bool TwoWire::begin(int sdaPin, int sclPin, uint32_t frequency)
         started = true;
         goto end;
     }
+    if (!allocateWireBuffer()) {
+        // failed! Error Message already issued
+        goto end;
+    }
     if(!initPins(sdaPin, sclPin)){
         goto end;
     }
     err = i2cInit(num, sda, scl, frequency);
     started = (err == ESP_OK);
 
 end:
+    freeWireBuffer();
```

Comment:
With this change its not working now. I did not see that it always run to `end:` even if all initialisations succeeds. We need to add a check to make it work correctly
```
if(!started) {
freeWireBuffer();
}
```

Reply:
yes, right, it falls through and causes the issue. :-)

Reply:
I don't like this `goto` way of doing it... but let's keep "the style" ;-)